### ISSUE TITLE
"fc.SIFormatValue" error

### ISSUE THREAD

**Issue by Alexustas** [MOARdV/AvionicsSystems#182]
in my pages  I'm using 
` . . .  $&$ fc.SIFormatValue(fc.MaxTWR(true),4,0,",",false,false);
fc.SIFormatValue(fc.MaxTWR(false),4,0,",",false,false)
`
and

`. . . $&$ fc.SIFormatValue(fc.CurrentTWR(),4,0,",",false,false);`

and after last release (0.20) I have an error
```
[ERR 14:08:42.349] [MdVTextMesh] Variable  fc.SIFormatValue(fc.MaxTWR(true),4,0,",",false,false) threw an exception

[ERR 14:08:42.349] [MASPage] Error configuring MASPage ASET_MFD2402_SYS_PROPULSION MAS_PAGE PERFORMANCE normal:

[ERR 14:08:42.350] [MASPage] System.ArgumentException: Error parsing variable "fc.SIFormatValue(fc.MaxTWR(true),4,0,",",false,false)" ---> System.ArgumentOutOfRangeException: Argument is out of range.
Parameter name: Value is too small or too big.
at System.Math.Round (double,int) <0x00086>
at AvionicsSystems.MASFlightComputerProxy.DoSIFormat (double,int,int,string,bool,bool) <0x0032f>
at AvionicsSystems.MASFlightComputerProxy.SIFormatValue (double,double,double,string,bool,bool) <0x0005d>
at (wrapper dynamic-method) AvionicsSystems.DynamicMethodFactory. (object,object[]) <0x00160>
at AvionicsSystems.MASFlightComputer/<>c__DisplayClass9f.<GenerateCallVariable>b__90 () <0x000a1>
at AvionicsSystems.StringVariable.Evaluate (bool) <0x00039>
at AvionicsSystems.StringVariable..ctor (string,System.Func`1<string>,bool,bool,AvionicsSystems.Variable/VariableType) <0x00075>
at AvionicsSystems.MASFlightComputer.GenerateCallVariable (AvionicsSystems.CodeGen.CallExpression) <0x00a98>
at AvionicsSystems.MASFlightComputer.GenerateVariable (AvionicsSystems.CodeGen.Expression) <0x00296>
at AvionicsSystems.MASFlightComputer.GetVariable (string,InternalProp) <0x002a8>
at AvionicsSystems.MASFlightComputer.RegisterVariableChangeCallback (string,InternalProp,System.Action`1<double>,bool) <0x0003e>

  --- End of inner exception stack trace ---
  at AvionicsSystems.MASFlightComputer.RegisterVariableChangeCallback (System.String variableName, .InternalProp prop, System.Action`1 callback, Boolean initializeNow) [0x00000] in <filename unknown>:0 
  at AvionicsSystems.VariableRegistrar.RegisterVariableChangeCallback (System.String name, System.Action`1 action, Boolean initializeNow) [0x00000] in <filename unknown>:0 
  at AvionicsSystems.MdVTextMesh.SetText (System.String text, Boolean immutable, Boolean preserveWhitespace, AvionicsSystems.MASFlightComputer comp, .InternalProp internalProp) [0x00000] in <filename unknown>:0 

[ERR 14:08:42.351] [AvionicsSystems] INITIALIZATION ERROR: Error configuring MASPage ASET_MFD2402_SYS_PROPULSION MAS_PAGE PERFORMANCE normal:

[ERR 14:08:42.353] [MdVTextMesh] Variable  fc.SIFormatValue(fc.CurrentTWR(),4,0,",",false,false) threw an exception

[ERR 14:08:42.353] [MASPage] Error configuring MASPage ASET_MFD2402_SYS_PROPULSION MAS_PAGE TWR BIG:

[ERR 14:08:42.354] [MASPage] System.ArgumentException: Error parsing variable "fc.SIFormatValue(fc.CurrentTWR(),4,0,",",false,false)" ---> System.ArgumentOutOfRangeException: Argument is out of range.
Parameter name: Value is too small or too big.
at System.Math.Round (double,int) <0x00086>
at AvionicsSystems.MASFlightComputerProxy.DoSIFormat (double,int,int,string,bool,bool) <0x0032f>
at AvionicsSystems.MASFlightComputerProxy.SIFormatValue (double,double,double,string,bool,bool) <0x0005d>
at (wrapper dynamic-method) AvionicsSystems.DynamicMethodFactory. (object,object[]) <0x00160>
at AvionicsSystems.MASFlightComputer/<>c__DisplayClass9f.<GenerateCallVariable>b__90 () <0x000a1>
at AvionicsSystems.StringVariable.Evaluate (bool) <0x00039>
at AvionicsSystems.StringVariable..ctor (string,System.Func`1<string>,bool,bool,AvionicsSystems.Variable/VariableType) <0x00075>
at AvionicsSystems.MASFlightComputer.GenerateCallVariable (AvionicsSystems.CodeGen.CallExpression) <0x00a98>
at AvionicsSystems.MASFlightComputer.GenerateVariable (AvionicsSystems.CodeGen.Expression) <0x00296>
at AvionicsSystems.MASFlightComputer.GetVariable (string,InternalProp) <0x002a8>
at AvionicsSystems.MASFlightComputer.RegisterVariableChangeCallback (string,InternalProp,System.Action`1<double>,bool) <0x0003e>

  --- End of inner exception stack trace ---
  at AvionicsSystems.MASFlightComputer.RegisterVariableChangeCallback (System.String variableName, .InternalProp prop, System.Action`1 callback, Boolean initializeNow) [0x00000] in <filename unknown>:0 
  at AvionicsSystems.VariableRegistrar.RegisterVariableChangeCallback (System.String name, System.Action`1 action, Boolean initializeNow) [0x00000] in <filename unknown>:0 
  at AvionicsSystems.MdVTextMesh.SetText (System.String text, Boolean immutable, Boolean preserveWhitespace, AvionicsSystems.MASFlightComputer comp, .InternalProp internalProp) [0x00000] in <filename unknown>:0 

```

in all other cases, the "fc.SIFormatValue"  function works fine

**Comment by MOARdV** [comment]
It is an error caused by divide-by-zero - the vessel mass has not been initialized when the props configure.  Easy fix.

**Comment by Alexustas** [comment]
and it back again (((

```
[ERR 14:23:38.724] [MdVTextMesh] Variable  fc.SIFormatValue(fc.Apoapsis(), 9, 0, ",", true, true) threw an exception

[ERR 14:23:38.725] [MASPage] Error configuring MASPage ASET_MFD2402_PFD_ASC/DES MAS_PAGE AD_Orb_NORMAL:

[ERR 14:23:38.726] [MASPage] System.ArgumentException: Error parsing variable "fc.SIFormatValue(fc.Apoapsis(), 9, 0, ",", true, true)" ---> System.ArgumentOutOfRangeException: Argument is out of range.
Parameter name: Value is too small or too big.
at System.Math.Round (double,int) <0x00086>
at AvionicsSystems.MASFlightComputerProxy.DoSIFormat (double,int,int,string,bool,bool) <0x0032f>
at AvionicsSystems.MASFlightComputerProxy.SIFormatValue (double,double,double,string,bool,bool) <0x0005d>
at (wrapper dynamic-method) AvionicsSystems.DynamicMethodFactory. (object,object[]) <0x00160>
at AvionicsSystems.MASFlightComputer/<>c__DisplayClass9f.<GenerateCallVariable>b__90 () <0x000a1>
at AvionicsSystems.StringVariable.Evaluate (bool) <0x00039>
at AvionicsSystems.StringVariable..ctor (string,System.Func`1<string>,bool,bool,AvionicsSystems.Variable/VariableType) <0x00075>
at AvionicsSystems.MASFlightComputer.GenerateCallVariable (AvionicsSystems.CodeGen.CallExpression) <0x00a98>
at AvionicsSystems.MASFlightComputer.GenerateVariable (AvionicsSystems.CodeGen.Expression) <0x00296>
at AvionicsSystems.MASFlightComputer.GetVariable (string,InternalProp) <0x002a8>
at AvionicsSystems.MASFlightComputer.RegisterVariableChangeCallback (string,InternalProp,System.Action`1<double>,bool) <0x0003e>

  --- End of inner exception stack trace ---
  at AvionicsSystems.MASFlightComputer.RegisterVariableChangeCallback (System.String variableName, .InternalProp prop, System.Action`1 callback, Boolean initializeNow) [0x00000] in <filename unknown>:0 
  at AvionicsSystems.VariableRegistrar.RegisterVariableChangeCallback (System.String name, System.Action`1 action, Boolean initializeNow) [0x00000] in <filename unknown>:0 
  at AvionicsSystems.MdVTextMesh.SetText (System.String text, Boolean immutable, Boolean preserveWhitespace, AvionicsSystems.MASFlightComputer comp, .InternalProp internalProp) [0x00000] in <filename unknown>:0 

[ERR 14:23:38.726] [AvionicsSystems] INITIALIZATION ERROR: Error configuring MASPage ASET_MFD2402_PFD_ASC/DES MAS_PAGE AD_Orb_NORMAL:

[ERR 14:23:38.727] [MdVTextMesh] Variable  fc.SIFormatValue(fc.Apoapsis()-fc.GetPersistent("FMS_Ap"),10,0,",",true,true) threw an exception

[ERR 14:23:38.728] [MASPage] Error configuring MASPage ASET_MFD2402_PFD_ASC/DES MAS_PAGE AD_Eng_SMALL:

[ERR 14:23:38.729] [MASPage] System.ArgumentException: Error parsing variable "fc.SIFormatValue(fc.Apoapsis()-fc.GetPersistent("FMS_Ap"),10,0,",",true,true)" ---> System.ArgumentOutOfRangeException: Argument is out of range.
Parameter name: Value is too small or too big.
at System.Math.Round (double,int) <0x00086>
at AvionicsSystems.MASFlightComputerProxy.DoSIFormat (double,int,int,string,bool,bool) <0x0032f>
at AvionicsSystems.MASFlightComputerProxy.SIFormatValue (double,double,double,string,bool,bool) <0x0005d>
at (wrapper dynamic-method) AvionicsSystems.DynamicMethodFactory. (object,object[]) <0x00160>
at AvionicsSystems.MASFlightComputer/<>c__DisplayClass9f.<GenerateCallVariable>b__90 () <0x000a1>
at AvionicsSystems.StringVariable.Evaluate (bool) <0x00039>
at AvionicsSystems.StringVariable..ctor (string,System.Func`1<string>,bool,bool,AvionicsSystems.Variable/VariableType) <0x00075>
at AvionicsSystems.MASFlightComputer.GenerateCallVariable (AvionicsSystems.CodeGen.CallExpression) <0x00a98>
at AvionicsSystems.MASFlightComputer.GenerateVariable (AvionicsSystems.CodeGen.Expression) <0x00296>
at AvionicsSystems.MASFlightComputer.GetVariable (string,InternalProp) <0x002a8>
at AvionicsSystems.MASFlightComputer.RegisterVariableChangeCallback (string,InternalProp,System.Action`1<double>,bool) <0x0003e>

  --- End of inner exception stack trace ---
  at AvionicsSystems.MASFlightComputer.RegisterVariableChangeCallback (System.String variableName, .InternalProp prop, System.Action`1 callback, Boolean initializeNow) [0x00000] in <filename unknown>:0 
  at AvionicsSystems.VariableRegistrar.RegisterVariableChangeCallback (System.String name, System.Action`1 action, Boolean initializeNow) [0x00000] in <filename unknown>:0 
  at AvionicsSystems.MdVTextMesh.SetText (System.String text, Boolean immutable, Boolean preserveWhitespace, AvionicsSystems.MASFlightComputer comp, .InternalProp internalProp) [0x00000] in <filename unknown>:0 

[ERR 14:23:38.729] [AvionicsSystems] INITIALIZATION ERROR: Error configuring MASPage ASET_MFD2402_PFD_ASC/DES MAS_PAGE AD_Eng_SMALL:
```

this error appears randomly (I mean it may appear, but it may not, with equal probability), when I go to o the Launch pad from VAB or when I switch to the vessel on the Launch pad from the KSC. if vessel is on orbit, this error does not appear.

this error was earlier and I already reported about[ it in this issue](https://github.com/MOARdV/AvionicsSystems/issues/90#issuecomment-373373121) and recorded the [video](https://youtu.be/i0R1-rgQ92o), then it disappeared for a long time. but now returned again

**Comment by MOARdV** [comment]
I now understand the problem.  There are two kinds of infinity, one for single-precision numbers, and one for double-precision.  If a number is single-precision infinity. it is not converted to double-precision infinity.  It is, instead, converted to a Big Number that is still valid.  I've added a test to SIFormat so very large numbers all become 0.

Fixed (hopefully, for the last time) on [DropBox](https://www.dropbox.com/s/qo4a5yxz3qa6065/AvionicsSystems-0.21.0b1.zip?dl=0).
